<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- Debug with gbd or valgrind -->
  <arg name="debug" default="" />

  <arg name="model" default="fixed_base_no_namespace"/>
  <arg name="description_name" default="smb_description"/>
  
  <arg name="node_name"                            default="smb_highlevel_controller" />
  <arg name="publish_ros"                          default="true"/>
  <!--  <arg name="joy_cmd_vel_topic"                    default="joy_manager/command_velocity"/>
  <arg name="cmd_vel_topic"                        default="$(arg node_name)/cmd_vel"/>
  <arg name="soft_emcy_stop_topic"                 default="soft_emcy_stop"/>
  <arg name="state_estimator_topic_smb"          default="state_estimator/smb_state"/>
  <arg name="joint_states_topic"                   default="/joint_states"/>
  <arg name="tf_topic"                             default="/tf"/>
  <arg name="hard_emcy_stop_topic"                 default="hard_emcy_stop"/>-->
  <!-- <arg name="ee_command_topic"                     default="$(arg node_name)/ee_state_command"/> -->

  <arg name="tf_minimal"                           default="false"/>
  <arg name="realsense_on_hand"                    default="false"/>
  <arg name="april_tag_detection"                  default="false"/>
  <arg name="tag_size"                             value="0.161"/>
  <arg name="time_step"                            default="0.0025"/>
  <arg name="simulation"                           default="false"/>
  <arg name="launch_joystick"                      default="true"/>
  <arg name="use_logitech_joy"                     default="true"/>
  <arg name="joy_interface_modules"                value="[]"/>
  <arg name="joy_modules"                          value="[]"/>
  <arg name="launch_prefix" default=""/>

  <!-- Cosmo sync name -->
  <param name="smb_highlevel_controller/sync_name" value="smb_controller_sync"/>
  <param name="smb_highlevel_controller/publish_ros" value="$(arg publish_ros)"/>
  <param name="smb_highlevel_controller/simulation" value="$(arg simulation)"/>

    <!-- High-level controller -->
  <node name="smb_highlevel_controller" pkg="smb_highlevel_controller" type="smb_highlevel_controller_node" 
      output="screen" launch-prefix="$(arg debug)">
    <rosparam file="$(find smb_highlevel_controller)/config/parameters.yaml" command="load" />
    <rosparam file="$(find smb_highlevel_controller)/config/controllers.yaml" command="load" />
    <rosparam file="$(find smb_highlevel_controller)/config/publishers.yaml" command="load" />
    <rosparam file="$(find smb_highlevel_controller)/config/subscribers.yaml" command="load" />
    <rosparam file="$(find smb_highlevel_controller)/config/services.yaml" command="load" />
    <param name="time_step" value="$(arg time_step)"/>
    <param name="logger/script" value="$(find smb_highlevel_controller)/config/logger.yaml"/>
    <param name="logger/sampling_window" value="120"/>
    <param name="standalone" value="false"/>
    <param name="num_spinners" value="4"/>
  </node>

  <!-- Low-level controller -->
  <!-- TODO This shouldnt be here! Make an smb_lpc package!  <node pkg="smb_lowlevel_controller" type="smb_lowlevel_controller_node" name="smb_lowlevel_controller" output="screen" launch-prefix="$(arg debug)">
      <rosparam command="load" file="$(find smb_lowlevel_controller)/config/ros.yaml"/>
      <param name="time_step"                   value="$(arg time_step)"/>
      <param name="standalone"                  value="false"/>
      <param name="publish_measurements"        value="$(arg publish_ros)"/>
      <param name="publishing_time_step"        value="$(arg time_step)"/>
      <param name="sync_name"                   value="smb_controller_sync"/>
      <param name="command_smb"                 value="true"/>
    </node> -->

</launch>
